eradication_success_assessment/get_required_effort.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+#
+#
+from scipy.stats import genextreme
+import numpy as np
+import pandas as pd
+import typer
+
+app = typer.Typer()
+
+
+@app.command()
+def get_required_effort(
+    name: str = "tests/data/camaras_trampa_erradicacion_rata_natividad.csv",
+    seed: bool = False,
+    n_bootstrapping: int = 30,
+):
+    capture_date = pd.to_datetime("2019-11-09")
+
+    datafile: str = name
+    data = pd.read_csv(datafile)
+    output = make_fit(data, capture_date, seed, n_bootstrapping)
+    print(output)
+
+
+def make_fit(data, capture_date, seed, n_bootstrapping):
+    if seed:
+        np.random.seed(3)
+    data_before_capture = _get_date_before_capture(data, capture_date)
+
+    effort_without_sighted = data_before_capture["Cantidad_de_trampas_activas"].sum()
+    data_before_capture["is_sighting"] = data_before_capture.Cantidad_de_avistamientos != 0
+    data_before_capture = _add_sighting(data_before_capture)
+    effort_per_sighting = data_before_capture.groupby("sighting")[
+        "Cantidad_de_trampas_activas"
+    ].sum()
+
+    n_effort_per_sighting = len(effort_per_sighting)
+    n_boostraping: int = n_bootstrapping
+    required_effort: np.array = np.zeros(n_boostraping)
+
+    success_probability: float = 0.99
+    for i in range(n_boostraping):
+        resampled_effort_per_sighting = np.random.choice(effort_per_sighting, n_effort_per_sighting)
+        fit = genextreme.fit(resampled_effort_per_sighting)
+        required_effort[i] = genextreme.ppf(success_probability, fit[0], fit[1], fit[2])
+
+    p_value_complement: float = 0.95
+    reported_effort = np.quantile(required_effort, p_value_complement).astype(int)
+    output: dict = {
+        "required_effort": reported_effort,
+        "success_probability": success_probability,
+        "significance_level": 1 - p_value_complement,
+        "effort_without_sighted": effort_without_sighted,
+    }
+    return output
+
+
+@app.command()
+def write_methodology():
+    print(
+        """
+    \subsection*{Análisis}
+Utilizamos la función de distribución de valores extremos generalizada (GEV, por sus siglas en 
+inglés) para modelar la probabilidad de éxito en la erradiación de rata a partir de las 
+observaciones en cámaras trampa. La GEV es una familia de funciones continuas de probabilidad que
+combina las funciones de Gumbel, Fréchet y Weibull, conocidas como funciones de distribución de 
+valores extremos de tipo I, II y III: 
```

Comment:
Por favor podrías quitar los espacios del final de las líneas?
```suggestion
Utilizamos la función de distribución de valores extremos generalizada (GEV, por sus siglas en
inglés) para modelar la probabilidad de éxito en la erradiación de rata a partir de las
observaciones en cámaras trampa. La GEV es una familia de funciones continuas de probabilidad que
combina las funciones de Gumbel, Fréchet y Weibull, conocidas como funciones de distribución de
valores extremos de tipo I, II y III:
```